File: backend/src/routes/finance.routes.ts
import { Router } from 'express'
import { requireAuth } from '../middlewares/auth.middleware'
import { listarTransacoes, criarTransacao, editarTransacao, deletarTransacao } from '../controllers/finance.controller'

const router = Router()

router.use(requireAuth)

router.get('/',       listarTransacoes)
router.post('/',      criarTransacao)
router.put('/:id',    editarTransacao)
router.delete('/:id', deletarTransacao)

export default router

File: backend/internal/finance/finance_routes.go
package finance

import (
	"github.com/gin-gonic/gin"
	"github.com/jackc/pgx/v5/pgxpool"

	"beeplanerofc/backend-go/internal/middleware"
	"beeplanerofc/backend-go/internal/supabaseauth"
)

// RegisterRoutes registra as rotas financeiras no grupo /api/finance. Todas exigem login.
func RegisterRoutes(router *gin.RouterGroup, db *pgxpool.Pool, authClient *supabaseauth.Client) {
	ctrl := NewController(db)

	router.Use(middleware.RequireAuth(db, authClient))

	router.GET("", ctrl.ListarTransacoes)
	router.POST("", ctrl.CriarTransacao)
	router.PUT("/:id", ctrl.EditarTransacao)
	router.DELETE("/:id", ctrl.DeletarTransacao)
}
